spline-shadow/client-ui/src/app/components/lineage/lineage-details/schema-details/schema-details.component.ts

```diff
@@ -42,17 +53,105 @@ export class SchemaDetailsComponent implements OnInit {
     return this.lineageGraphService.getColorFromOperationType(this.getType())
   }
 
-  getType(): string {
+  getType(property?: any): string {
+    if (property) {
```

Comment:
Just a hint - wouldn't the following be simpler and more readable?
`return (property || this.detailsInfo)._typeHint.split('.').pop() || undefined`

Reply:
correction:
`return ((property || this.detailsInfo)._typeHint || '').split('.').pop() || undefined`
or using LoDash:
`return _.split((property || this.detailsInfo)._typeHint, '.').pop() || undefined`

Reply:
To me the original line is simpler. Or I would use the LoDash version, but still  extract extra variable or function to reduce complexity of the line.

Reply:
could be a few lines, yes. My point was to get rid of that branching. Let me explain. The original version uses `if`-block combined with a ternary operator. So we have three branches in total, two of which only differs in an object on that they are applied.
In my version the intention looks strait forward - you take one or another variable, take a certain property, process it and return a result if any.